### PR TITLE
🚀 [Milestone] Add Reducers and Actions :books:

### PR DESCRIPTION
**Pull request by jicamargo**
# 🚀 [Milestone] Add Reducers and Actions 📚

Description:
This PR adds Redux Toolkit to the project and implements the necessary reducers and actions to manage books and categories. Here are the key features and requirements addressed in this milestone:

📋 Requirements:
- Setup Redux Toolkit within the project.
- Create a directory for Redux logic.
- Configure a Redux store.
- Implement a slice of state for books, including an array of books, and reducers for adding and removing books.
- Implement a slice of state for categories, including an array of categories, and a reducer that always returns "Under construction" as the initial state.

File Structure:
- Introduced a "redux" directory to house all Redux-related files.
- Created "books" and "categories" directories within "redux" to organize slice files.
- Implemented "booksSlice.js" containing the book-related state and reducers.
- Implemented "categoriesSlice.js" containing the category-related state and the "Under construction" reducer.

🎉 Please review the changes and provide your valuable feedback.

Thank you! 😊

### CODE REVIEW

File: src/redux/categories/categoriesSlice.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+const initialState = 'Under fconstruction'; // Set the initial state as "Under construction"
+const categoriesSlice = createSlice({
+  name: 'categories',
+  initialState,
+  reducers: {
+    // reducers specific to categories if needed
+  },
+});
+
```
Comment (Gambit142) [inline src/redux/categories/categoriesSlice.js]: - [x] Great work creating the **_categorySlice_**. The `initialState` should be equal to an empty array and then you ought to create a reducer called `checkStatus` that returns the string `'Under Construction'` if the `action.payload` is equal to `'Under Construction'`. Kindly implement it this way. 👍 💻

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @jicamargo 👋, 

Good job so far!

### Highlights
- All linters checks passed ✔️ 
- Installed `@reduxjs/toolkit` ✔️ 
- Installed `react-redux` ✔️
- Set up the Redux store properly using `configureStore` ✔️ 
- Set up reducers and actions using `createSlice` ✔️ 

You still need to work on some issues to go to the next project, but you are almost there!

## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend considering them as they can improve your code._

You can also consider the following:

N/A

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### Please tag me in your question so I can receive the notification. You can tag me by typing @Gambit142

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: src/redux/categories/categoriesSlice.js (hunk 1 above)
Comment (jicamargo) [reply src/redux/categories/categoriesSlice.js]: Thank you @Gambit142. I'll follow your suggestion.

**Review by bienvenuushindi — APPROVED** [review]
# STATUS: APPROVED 🟢 

## Hi @jicamargo   👋, 

Well done implementing the requested changes 👏
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



### Highlights 🎯 
:+1: No linter errors✔️
:+1: Good UI design✔️
:+1: Good PR title

### Optional Suggestions



**_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._**

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request closed without merging**